woocommerce/woocommerce-gateway-stripe
# Add onboarding payment gateway setup methods

# Changes proposed in this Pull Request:

Part of - woocommerce/woocommerce#32258

Adds the methods needed to use the config-based setup for the payment gateway during onboarding.

<img width="707" alt="Screen Shot 2021-05-27 at 11 31 15 AM" src="https://user-images.githubusercontent.com/10561050/119854694-49cba780-bedf-11eb-99f2-55d627b301b9.png">
<img width="705" alt="Screen Shot 2021-05-27 at 11 31 11 AM" src="https://user-images.githubusercontent.com/10561050/119854697-4a643e00-bedf-11eb-845d-28870d3ba3e9.png">


# Testing instructions

1. Install and activate this plugin in your `plugins/` directory matching the wp.org slug.  `woocommerce-gateway-stripe`
2. Install and activate the development version of [WooCommerce Admin](https://github.com/woocommerce/woocommerce-admin) ( `git clone https://github.com/woocommerce/woocommerce-admin.git && cd woocommerce-admin && composer install && npm i && npm run dev` ).
3. Make sure a qualifying Stripe country is selected for the store base location in WooCommerce general settings and that the "cbd" industry is not selected during onboarding.
2. Navigate to the Stripe gateway in the payments task `/wp-admin/admin.php?page=wc-admin&task=payments`
3. Fill out the required setting fields.
4. Make sure that the Stripe gateway is moved to the "Enabled payment gateways" section.
5. Make sure that settings are persisted on page refresh.
6. When adding testing and publishable keys that start with `test_`, test mode should be toggled on.  Currently this does not mark setup as "complete" but I think this open to interpretation and we can update accordingly.
